lib/resourceread/machineconfig.go

```diff
@@ -17,13 +17,22 @@ func init() {
 	}
 }
 
-// ReadMachineConfigV1OrDie reads MachineConfig object from bytes. Panics on error.
+// ReadMachineConfigV1 reads raw MachineConfig object from bytes. Returns MachineConfig and error.
+func ReadMachineConfigV1(objBytes []byte) (*mcfgv1.MachineConfig, error) {
+	mc, err := runtime.Decode(mcfgCodecs.UniversalDecoder(mcfgv1.SchemeGroupVersion), objBytes)
+	if err == nil {
```

Comment:
```go
if err !=nil {
    return nil, err
}
mc, ok := obj.(*mcfgv1.MachineConfig)
if !ok {
    return nil, <error describing this>
}
return mc, nil
```

Reply:
Thanks for the feedback...I've hardened the logic and repushed.